web/app/view/ReportController.js

```diff
@@ -102,9 +102,10 @@ Ext.define('Traccar.view.ReportController', {
                         to: to.toISOString()
                     }
                 });
-            } else if (button.reference === 'csvButton') {
+            } else if (button.reference === 'xlsxButton') {
                 url = this.getView().getStore().getProxy().url;
-                this.downloadCsv(url, {
+                accept = 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet';
+                this.downloadFile(url, accept, {
```

Comment:
I was thinking to make universal function to download files, but you are right, will change it in future if it will be necessary.